weierstrass/src/utils.rs

```diff
@@ -0,0 +1,26 @@
+use generic_array::typenum::Unsigned;
+use crate::{DoubleWord, Word, WordWidth};
+
+/// Computes `a + b + carry`, returning the result along with the new carry.
+#[inline(always)]
+pub(crate) const fn adc(a: Word, b: Word, carry: Word) -> (Word, Word) {
+    let ret = (a as DoubleWord) + (b as DoubleWord) + (carry as DoubleWord);
+    (ret as Word, (ret >> WordWidth::USIZE) as Word)
+}
+
+/// Computes `a - (b + borrow)`, returning the result along with the new borrow.
+#[inline(always)]
+pub(crate) const fn sbb(a: Word, b: Word, borrow: Word) -> (Word, Word) {
+    let (a, b) = (a as DoubleWord, b as DoubleWord);
+    let t = (borrow >> (WordWidth::USIZE - 1)) as DoubleWord;
+    let ret = a.wrapping_sub(b + t);
+    (ret as Word, (ret >> WordWidth::USIZE) as Word)
+}
+
+/// Computes `a + (b * c) + carry`, returning the result along with the new carry.
+#[inline(always)]
+pub(crate) const fn mac(a: Word, b: Word, c: Word, carry: Word) -> (Word, Word) {
+    let (a, b, c) = (a as DoubleWord, b as DoubleWord, c as DoubleWord);
+    let ret = a + b * c + (carry as DoubleWord);
+    (ret as Word, (ret >> WordWidth::USIZE) as Word)
+}
```

Comment:
FYI, there are some equivalents of these in:

https://docs.rs/elliptic-curve/0.6.6/elliptic_curve/util/index.html

Reply:
Yes, but those functions work with `u64`, while these ones with words (i.e. will use `u32` on 32-bit targets).

Reply:
They work with both, actually! There are 32-bit and 64-bit equivalents of each.

The `Word`-based approach is interesting though, and the only reason I haven't done such gating is because `p256` presently only has a 64-bit backend (`k256` has both 32-bit and 64-bit).